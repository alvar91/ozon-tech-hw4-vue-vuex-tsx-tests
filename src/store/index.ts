import Vue from "vue";
import Vuex from 'vuex';
import { createVuexStore } from "vuex-simple";

// @ts-ignore
import { MainModule } from "./modules/main/main.ts";

Vue.use(Vuex);

const store = createVuexStore(new MainModule());

export default store;
