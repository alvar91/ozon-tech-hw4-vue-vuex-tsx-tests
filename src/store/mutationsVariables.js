export const FETCH_USERS_ACTION = "FETCH_USERS_ACTION";
export const FETCH_USERS = "FETCH_USERS";
export const SET_LOADING = "SET_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SET_FILTER_KEYWORD = "SET_FILTER_KEYWORD";
export const SET_FORM_INPUT = "SET_FORM_INPUT";
export const SET_FORM_CHANGE = "SET_FORM_CHANGE";
export const SET_FORM_RADIO = "SET_FORM_RADIO";
