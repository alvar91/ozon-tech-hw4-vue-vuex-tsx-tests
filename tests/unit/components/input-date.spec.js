import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import InputDate from "@/components/input-date.vue";

import { mockUserId, mockUser, mockUserAge } from "../../mock/mockUser";

import {
  FETCH_USERS_ACTION,
  SET_CURRENT_USER,
} from "@/store/mutationsVariables";

import store from "@/store/index";

const localVue = createLocalVue();
localVue.use(Vuex);

const { age } = mockUser;
describe("InputDate.vue", () => {
  let wrapper;

  beforeEach(async () => {
    await store.dispatch(FETCH_USERS_ACTION);
    store.commit(SET_CURRENT_USER, mockUserId);
  });

  it("should correctly renders markup with default props", () => {
    wrapper = mount(InputDate, {
      localVue,
      propsData: {
        title: "Дата рождения",
        name: "age",
        value: { age },
      },
      store,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should correctly input data and change the store", () => {
    wrapper = mount(InputDate, {
      localVue,
      propsData: {
        title: "Дата рождения",
        name: "age",
        value: { age },
      },
      store,
    });

    const input = wrapper.find(".card__input");
    input.element.value = mockUserAge;
    input.trigger("input");

    expect(store.getters.currentUser.age).toBe(mockUserAge);
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
