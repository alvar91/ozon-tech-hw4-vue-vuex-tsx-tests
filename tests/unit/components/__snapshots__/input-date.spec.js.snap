// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDate.vue should correctly renders markup with default props 1`] = `<label class="card__field"><span class="card__title">Дата рождения</span><input type="date" name="age" min="1900-01-01" class="card__input card__input--date"></label>`;
